server/src/main/java/org/opensearch/OpenSearchException.java

```diff
@@ -594,16 +594,14 @@ public static void generateFailureXContent(XContentBuilder builder, Params param
 
         // Render the exception with a simple message
         if (detailed == false) {
-            String message = "No OpenSearchException found";
             Throwable t = e;
             for (int counter = 0; counter < 10 && t != null; counter++) {
                 if (t instanceof OpenSearchException) {
-                    message = t.getClass().getSimpleName() + "[" + t.getMessage() + "]";
                     break;
                 }
                 t = t.getCause();
             }
-            builder.field(ERROR, message);
+            builder.field(ERROR, ExceptionsHelper.summaryMessage(t));
```

Comment:
```suggestion
            builder.field(ERROR, ExceptionsHelper.summaryMessage(e));
```
`t` is null, which results in a more confusing `Internal error` exception even when a useful `IllegalArgumentException` is thrown. `e` should be passed to `summaryMessage` so a more useful message is thrown. This already resulted in a [reproducible test failure](https://build.ci.opensearch.org/job/gradle-check/4166/) which was fixed in #4702.

Reply:
@nknize I think the logic behind this code is to find the `OpenSearchException` down the cause chain (which is `t`). May be in this case we should do:

```
builder.field(ERROR, ExceptionsHelper.summaryMessage(t != null ? t : e));
```

?

Reply:
Thanks for reviewing. I will do this fix

Reply:
I agree with the suggested change from @reta . The original logic is to find the `OpenSearchException` down the cause chain, and show its error message.

Reply:
The `ExceptionsHelper.summaryMessage(e)` version is what [exists on main currently](https://github.com/opensearch-project/OpenSearch/blob/f0511311e3166cda7ef47fc5ba7e45b61240b94b/server/src/main/java/org/opensearch/OpenSearchException.java#L591). We should probably keep them consistent and then follow up with a change on main to do the `ExceptionsHelper.summaryMessage(t != null ? t : e)` version (and then backport that PR as well).

Reply:
@andrross Make sense, thank you! 😄